app/controllers/calagator/admin_controller.rb

```diff
@@ -1,27 +1,41 @@
 module Calagator
 
-class AdminController < Calagator::ApplicationController
-  require_admin
+  class AdminController < Calagator::ApplicationController
+    require_admin
 
-  def index
-  end
+    def index
+    end
 
-  def events
-    @events = Event.future
-  end
+    def events
+      if params[:query].blank?
+        @events = Event.future
+      else
+        @search = Event::Search.new(params)
+        @admin_query = params[:query]
 
-  def lock_event
-    @event = Event.find(params[:event_id])
-    if @event.locked?
-      @event.unlock_editing!
-      flash[:success] = "Unlocked event #{@event.title} (#{@event.id})"
-    else
-      @event.lock_editing!
-      flash[:success] = "Locked event #{@event.title} (#{@event.id})"
+        # setting @events so that we can reuse the index atom builder
+        @events = @search.events
```

Comment:
Probably can ditch this, because I doubt we are publishing an atom feed of the administration section.